docs: use a processor to transform the operator decision tree data

tmair opened the pull request:
<!--
Thank you very much for your pull request!

If your PR is the addition of a new operator, please make sure all these boxes are ticked with an x:

- [ ] Add the operator to Rx
- [ ] It must have a `-spec.ts` tests file covering the canonical corner cases, with marble diagram tests
- [ ] The spec file should have a type definition test at the end of the spec to verify type definition for various use cases
- [ ] The operator must be documented in JSDoc style in the implementation file, including also the PNG marble diagram image
- [ ] The operator should be listed in `apps/rxjs.dev/content/guide/operators.md` in a category of operators
- [ ] The operator should also be documented. See [Documentation Guidelines](../CONTRIBUTING.md).
- [ ] You may need to update `MIGRATION.md` if the operator differs from the corresponding one in RxJS v4
-->

**Description:**
This PR moves the generation of the decision tree into a dgeni processor, so that the decision tree will be generated as part of the documentation generation.

<!--
If this PR introduces a breaking change, it must contain a notice for it to be included in the CHANGELOG.
-->

**Related issue (if exists):**
None

benlesh commented:
Hey @tmair ! I really appreciate all of this good work. Generally, I'm not the one that reviews anything to do with the docs app, so I've been deferring to @jakovljevic-mladen and @niklas-wortmann for that. This seems like a nice add though.

niklas-wortmann reviewed (approved):
LGTM thanks for adjusting this

jakovljevic-mladen reviewed (changes requested):
@tmair, this PR is awesome, thank you a lot for merging Operator Decision Tree generator to the tools that already exist.

I have one comment regarding a file rename; everything else looks great.

tmair commented on apps/rxjs.dev/tools/transforms/config.js:
```diff
@@ -12,6 +12,7 @@ const DOCS_OUTPUT_PATH = resolve(OUTPUT_PATH, 'docs');
 const API_SOURCE_PATH = resolve(PROJECT_ROOT, 'packages/rxjs/src');
 const MARBLE_IMAGES_PATH = resolve(SRC_PATH, 'assets/images/marble-diagrams');
 const MARBLE_IMAGES_WEB_PATH = 'assets/images/marble-diagrams';
+const DECISION_TREE_PATH = resolve(CONTENTS_PATH, 'tree.yml');
```
Yes, that is a good remark. I renamed the file.

tmair commented:
> I have one comment regarding a file rename; everything else looks great. Also, conflicts need to be resolved.

I rebased the PR and addressed the comment about the renaming of the file.

jakovljevic-mladen reviewed (approved):
LGTM. Thanks a lot for making these changes @tmair 🎉
